File: solutions/easy/905.Sort_Array_By_Parity/905.Sort_Array_By_Parity.ts
function sortArrayByParity(nums: number[]): number[] {
  let k = 0
  for(let i = 0; i < nums.length; i++) {
      if(nums[i] % 2 === 0) {
          let temp = nums[k]
          nums[k] = nums[i]
          nums[i] = temp
          k++
      }
  }
  return nums
};

function sortArrayByParitySecond(nums: number[]): number[] {
  let k = 0
  for(let i = 0; i < nums.length; i++) {
      if(nums[i] % 2 === 0) {
          [nums[k], nums[i]] = [nums[i], nums[k]]
          k++
      }
  }
  return nums
};

File: solutions/easy/905.Sort_Array_By_Parity/905.Sort_Array_By_Parity.py
from ast import List


class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        k = 0
        for i in range(len(nums)):
            if nums[i] % 2 == 0:
                nums[k], nums[i] = nums[i], nums[k]
                k += 1
        return nums
